addons/mail/static/src/core/common/message_model.js

```diff
@@ -462,28 +462,43 @@ export class Message extends Record {
     previewText = fields.Html("", {
         /** @this {import("models").Message} */
         compute() {
+            let messageBody = "";
             if (!this.hasOnlyAttachments) {
-                return this.inlineBody || this.subtype_id?.description;
+                messageBody = this.inlineBody || this.subtype_id?.description;
+            } else {
+                const attachments = this.attachment_ids;
+                switch (attachments.length) {
+                    case 1:
+                        messageBody = attachments[0].previewName;
+                        break;
+                    case 2:
+                        messageBody = _t("%(file1)s and %(file2)s", {
+                            file1: attachments[0].previewName,
+                            file2: attachments[1].previewName,
+                            count: attachments.length - 1,
+                        });
+                        break;
+                    default:
+                        messageBody = _t("%(file1)s and %(count)s other attachments", {
+                            file1: attachments[0].previewName,
+                            count: attachments.length - 1,
+                        });
```

Comment:
This phrase is not correctly built with the number for translation, already before this PR.

Numbers need the very minimum at least the case of 0 (already covered above), the case for 1, and the case for many. There are many other exceptions to get all languages right but 0 1 and many is really the minimum to even get English and French right.

"1 other attachments" (plural) is not correct, same in french were we need to plural both words.

This can be done as a follow up.